modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -1150,42 +1150,59 @@ private void processOperation(String resourcePath, String httpMethod, Operation
 
         final Map<String, SecurityScheme> securitySchemes = openAPI.getComponents() != null ? openAPI.getComponents().getSecuritySchemes() : null;
         final List<SecurityRequirement> globalSecurities = openAPI.getSecurity();
-        for (Tag tag : tags) {
-            try {
-                CodegenOperation codegenOperation = config.fromOperation(resourcePath, httpMethod, operation, path.getServers());
-                codegenOperation.tags = new ArrayList<>(tags);
-                config.addOperationToGroup(config.sanitizeTag(tag.getName()), resourcePath, operation, codegenOperation, operations);
 
-                List<SecurityRequirement> securities = operation.getSecurity();
-                if (securities != null && securities.isEmpty()) {
-                    continue;
-                }
+        // Find the first possible tag
+        Tag tag = null;
+        for (Tag tmpTag : tags) {
+            List<SecurityRequirement> securities = operation.getSecurity();
+            if (securities != null && securities.isEmpty()) {
+                continue;
+            }
```

Comment:
It looks to me like the 4 rows above are not needed, as securities are not dependent on tags, but are the same for the operation. It also seems like no tag would be selected if `securities != null && securities.isEmpty()` is `true`.

Possibly something like this instead of the whole for loop:?
```java
if (tags.empty()) {
    LOGGER.warn("At least one tag should be specified for operation '{}', resource {} '{}'. Ignoring operation", operation.getOperationId(), httpMethod, resourcePath);
    return;
}
Tag tag = tags.get(0);
```